spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -407,13 +407,19 @@ public Object postProcessAfterInitialization(final Object bean, final String bea
 					this.logger.debug(() -> annotatedMethods.size() + " @KafkaListener methods processed on bean '"
 							+ beanName + "': " + annotatedMethods);
 				}
+				Set<Method> methodsWithHandler = MethodIntrospector.selectMethods(targetClass,
+						(ReflectionUtils.MethodFilter) method ->
+								AnnotationUtils.findAnnotation(method, KafkaHandler.class) != null);
+				boolean hasMethodLevelKafkaHandlerAnnotation = !methodsWithHandler.isEmpty();
 				if (hasClassLevelListeners) {
-					Set<Method> methodsWithHandler = MethodIntrospector.selectMethods(targetClass,
-							(ReflectionUtils.MethodFilter) method ->
-									AnnotationUtils.findAnnotation(method, KafkaHandler.class) != null);
 					List<Method> multiMethods = new ArrayList<>(methodsWithHandler);
 					processMultiMethodListeners(classLevelListeners, multiMethods, targetClass, bean, beanName);
 				}
+
+				if (!hasMethodLevelListeners && hasClassLevelListeners &&
+					!hasMethodLevelKafkaHandlerAnnotation) {
+					throw new IllegalStateException("No kafka listener methods found on bean type.");
```

Comment:
Sorry, I probably was not clear enough .
I mean like this:
```suggestion
					throw new IllegalStateException("No Kafka listener methods in bean: " + bean);
```

Reply:
I think we should not take into a consideration that `BeanCreationException`. Might be the fact that it is not thrown upstream.
Why my suggestion does not work for you then?